### PR TITLE
Added SubscribeButton RPC processing

### PR DESCRIPTION
**Pull request by BSolonenko**
For SubscribeButton RPC, the ability respond with different resultCodes
for different applications was added.

### CODE REVIEW

File: ffw/ButtonsRPC.js [hunk 1]
```diff
@@ -129,136 +130,137 @@ FFW.Buttons = FFW.RPCObserver.create(
...
+            );
+          } else {
+            this.sendError(
+              result_code, request.id, request.method, result_info
+            );
+          }
+          break;
+        }
+        case 'Buttons.GetCapabilities' : {
+          capabiliti = function(name){
```
Comment (AKalinich-Luxoft) [inline ffw/ButtonsRPC.js]: @BSolonenko `capability`

File: ffw/ButtonsRPC.js [hunk 2]
```diff
@@ -129,136 +130,137 @@ FFW.Buttons = FFW.RPCObserver.create(
...
-        } else {
-          this.sendError(
-            result_code, request.id, request.method, result_info
-          );
-        }
+    isButtonSubscribed: function(appID, buttonName){
+      if(null == this.subscribedButtons[appID]){
+        return false;
+      }
+      if(null ==  this.subscribedButtons[appID][buttonName]){
```
Comment (AKalinich-Luxoft) [inline ffw/ButtonsRPC.js]: @BSolonenko extra space

File: ffw/ButtonsRPC.js [hunk 3]
```diff
@@ -129,136 +130,137 @@ FFW.Buttons = FFW.RPCObserver.create(
...
+                capabiliti("PRESET_7"),
+                capabiliti("PRESET_8"),
+                capabiliti("PRESET_9"),
+                capabiliti("OK"),
+                capabiliti("PLAY_PAUSE"),
+                capabiliti("SEEKLEFT"),
+                capabiliti("SEEKRIGHT"),
+                capabiliti("TUNEUP"),
+                capabiliti("TUNEDOWN"),
+                capabiliti("CUSTOM_BUTTON")
```
Comment (AKalinich-Luxoft) [inline ffw/ButtonsRPC.js]: @BSolonenko is it possible to generate this long list over iterating elements in 'FFW.RPCHelper.SubscribeButton'?

File: app/view/settings/RPCControls/RPCControlConfigView.js [hunk 4]
```diff
@@ -169,6 +170,47 @@ SDL.RPCControlConfigView = Em.ContainerView.create(
...
+            childViews: [
+              'header'
+            ],
+            header:  SDL.Label.extend({
+              elementId: 'header',
+              classNames: 'header',
+              content: 'header'
+            }),
+            isFirstInit: true,
+            initSubscribeButtonViev: function(){
```
Comment (AKalinich-Luxoft) [inline app/view/settings/RPCControls/RPCControlConfigView.js]: @BSolonenko and again, is it possible to move this logic into `init` function? In this case you will be sure that it was inited once and don't need this isFirstInit flag

File: ffw/ButtonsRPC.js (hunk 1 above)
Comment (BSolonenko) [reply ffw/ButtonsRPC.js]: @AKalinich-Luxoft  fixed in https://github.com/smartdevicelink/sdl_hmi/pull/126/commits/3fa0234004335e9b335fe62a02a68bbb5e2ffed7

File: ffw/ButtonsRPC.js (hunk 2 above)
Comment (BSolonenko) [reply ffw/ButtonsRPC.js]: @AKalinich-Luxoft  fixed in https://github.com/smartdevicelink/sdl_hmi/pull/126/commits/3fa0234004335e9b335fe62a02a68bbb5e2ffed7

File: ffw/ButtonsRPC.js (hunk 3 above)
Comment (BSolonenko) [reply ffw/ButtonsRPC.js]: @AKalinich-Luxoft  fixed in https://github.com/smartdevicelink/sdl_hmi/pull/126/commits/3fa0234004335e9b335fe62a02a68bbb5e2ffed7

File: app/view/settings/RPCControls/RPCControlConfigView.js (hunk 4 above)
Comment (BSolonenko) [reply app/view/settings/RPCControls/RPCControlConfigView.js]: @AKalinich-Luxoft  fixed in https://github.com/smartdevicelink/sdl_hmi/pull/126/commits/3fa0234004335e9b335fe62a02a68bbb5e2ffed7

File: app/view/settings/RPCControls/RPCControlConfigView.js [hunk 5]
```diff
@@ -169,6 +170,47 @@ SDL.RPCControlConfigView = Em.ContainerView.create(
                     }
                   )
                 }
-              )
+              ),
```
[the remark this answers is not in the document]
Comment (BSolonenko) [reply app/view/settings/RPCControls/RPCControlConfigView.js]: @ValeriiMalkov  fixed in https://github.com/smartdevicelink/sdl_hmi/pull/126/commits/3fa0234004335e9b335fe62a02a68bbb5e2ffed7

File: app/view/settings/RPCControls/RPCControlConfigView.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (BSolonenko) [reply app/view/settings/RPCControls/RPCControlConfigView.js]: @ValeriiMalkov  fixed in https://github.com/smartdevicelink/sdl_hmi/pull/126/commits/3fa0234004335e9b335fe62a02a68bbb5e2ffed7

File: app/view/settings/RPCControls/RPCControlConfigView.js [hunk 6]
```diff
@@ -169,6 +170,47 @@ SDL.RPCControlConfigView = Em.ContainerView.create(
...
+                          elementId: param + 'Select',
+                          classNames: 'select',
+                          contentBinding: 'SDL.SDLModel.data.resultCodes',
+                          valueBinding: 'FFW.RPCHelper.SubscribeButton.' + param
+                        })  
+                    }));
+                }
+            }
+          })     
     }
```
[the remark this answers is not in the document]
Comment (BSolonenko) [reply app/view/settings/RPCControls/RPCControlConfigView.js]: @ValeriiMalkov  fixed in https://github.com/smartdevicelink/sdl_hmi/pull/126/commits/3fa0234004335e9b335fe62a02a68bbb5e2ffed7

File: ffw/ButtonsRPC.js [hunk 7]
```diff
@@ -129,136 +130,137 @@ FFW.Buttons = FFW.RPCObserver.create(
...
+        case 'Buttons.GetCapabilities' : {
+          capabiliti = function(name){
+              return {
+                  'name': name,
+                  'shortPressAvailable': true,
+                  'longPressAvailable': true,
+                  'upDownAvailable': true
+              };
+          };
+          // send repsonse
```
[the remark this answers is not in the document]
Comment (BSolonenko) [reply ffw/ButtonsRPC.js]: @ValeriiMalkov  fixed in https://github.com/smartdevicelink/sdl_hmi/pull/126/commits/3fa0234004335e9b335fe62a02a68bbb5e2ffed7

**Pull request merged**